monitor/utils/message.js

```diff
@@ -64,9 +67,22 @@ const eventWithoutReference = otherSideEvents => e =>
   otherSideEvents.filter(a => a.referenceTx === e.referenceTx && a.recipient === e.recipient && a.value === e.value)
     .length === 0
 
+const unclaimedHomeToForeignRequests = () => {
+  if (MONITOR_HOME_TO_FOREIGN_ALLOWANCE_LIST) {
+    const allowanceList = readAccessListFile(MONITOR_HOME_TO_FOREIGN_ALLOWANCE_LIST)
+    return e => !allowanceList.includes(e.recipient.toLowerCase())
```

Comment:
The similar functionality in the oracle also depends on the sender -- whether the sender of transaction is in AL/BL. I understand that it could impact the duration how the monitor handles the bridge operations. Could we estimate how duration will be increased? In order to speed up the process does it makes sense to cache the correspondence between a sender and a transaction as so monitor workers do not need to get the transaction data again and again?

Reply:
Well, if the sender of a transaction will be cached, so it is required to fetch it only once, than nothing bad should happen. Assuming that we have 1000 tx per day, each of them will be marked as unclaimed, each request takes ~300ms to execute, in total we have 300 seconds per day, or 13 seconds per hour.
I don't see a big problem here, so everything should be fine.